### PR TITLE
Decoration (issue markers)

### PR DESCRIPTION
**Pull request by JanKoehnlein**

### CODE REVIEW

File: src/features/decoration/decoration-placer.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/********************************************************************************
+ * Copyright (c) 2017-2018 TypeFox and others.
```
Comment (spoenemann) [inline src/features/decoration/decoration-placer.ts]: `Copyright (c) 2018`

File: src/features/decoration/decoration-placer.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import { SEdge } from "../../graph/sgraph";
+import { isSizeable } from "../bounds/model";
+
+@injectable()
+export class DecorationPlacer implements IVNodeDecorator {
+    decorate(vnode: VNode, element: SModelElement): VNode {
+        if (isDecoration(element)) {
+            const position = this.getPosition(element);
+            const translate = 'translate(' + position.x + ', ' + position.y + ')';
+            setAttr(vnode, 'transform', translate);
```
Comment (spoenemann) [inline src/features/decoration/decoration-placer.ts]: Does this work with all kinds of SVG nodes?

File: src/features/decoration/decoration-placer.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        }
+        return vnode;
+    }
+
+    protected getPosition(element: SModelElement & Decoration): Point {
+        if (element instanceof SChildElement && element.parent instanceof SEdge) {
```
Comment (spoenemann) [inline src/features/decoration/decoration-placer.ts]: Generalize to `isRoutable(element.parent)`

File: src/features/decoration/di.config.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+/********************************************************************************
+ * Copyright (c) 2017-2018 TypeFox and others.
```
Comment (spoenemann) [inline src/features/decoration/di.config.ts]: `Copyright (c) 2018`

File: src/features/decoration/di.config.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+import { configureModelElement } from "../../base/views/view";
+import { ContainerModule } from "inversify";
+import { SIssueMarker } from "./model";
+import { IssueMarkerView } from "./views";
+import { TYPES } from "../../base/types";
+import { DecorationPlacer } from "./decoration-placer";
+
+const decorationModule = new ContainerModule(bind => {
+    configureModelElement({bind}, 'marker', SIssueMarker, IssueMarkerView);
```
Comment (spoenemann) [inline src/features/decoration/di.config.ts]: Better choose a more specific type string such as `issue-marker`?

File: src/features/decoration/model.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
+/********************************************************************************
+ * Copyright (c) 2017-2018 TypeFox and others.
```
Comment (spoenemann) [inline src/features/decoration/model.ts]: `Copyright (c) 2018`

File: src/features/decoration/views.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
+/********************************************************************************
+ * Copyright (c) 2017-2018 TypeFox and others.
```
Comment (spoenemann) [inline src/features/decoration/views.tsx]: `Copyright (c) 2018`

File: src/features/decoration/decoration-placer.ts (hunk 2 above)
Comment (JanKoehnlein) [reply src/features/decoration/decoration-placer.ts]: Yes, AFAICS.

File: src/features/decoration/di.config.ts (hunk 5 above)
Comment (JanKoehnlein) [reply src/features/decoration/di.config.ts]: I like marker better (coming from Eclipse). Note that we usually allow a `:` to separate base type and sub type. I'd find introducing a `-` a bit confusing.

File: src/features/decoration/decoration-placer.ts (hunk 3 above)
Comment (JanKoehnlein) [reply src/features/decoration/decoration-placer.ts]: What else is routable if not an edge?

Comment (spoenemann) [reply src/features/decoration/decoration-placer.ts]: It's more a dependency issue. We have to be careful about dependencies from src/graph to src/features and back.

**Pull request merged**